compiler/generate_build_properties.gradle

```diff
@@ -34,3 +34,8 @@ def generateBuildProperties = project.tasks.register('generateBuildProperties')
 tasks.named("compileKotlin").configure {
   it.dependsOn generateBuildProperties
 }
+
+// So Gradle stops complaining about the missing dependency
+tasks.named("runKtlintCheckOverTestSourceSet").configure {
+  it.dependsOn generateBuildProperties
```

Comment:
I believe a `mustRunAfter` is good enough here. The benefit is that it doesn't make the other task run unnecessarily, but still teaches Gradle about the ordering.